core/src/main/scala/org/apache/gearpump/jarstore/JarFileContainer.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gearpump.jarstore
+
+import java.io.File
+
+/**
+ * The JarStore is a distributed store
+ * User can use this jar store to submit application jars
+ */
+trait JarFileContainer extends Serializable {
+
+  def copyFromLocal(localFile : File) : Unit
+
+  def copyToLocalFile(localPath : File) : Unit
```

Comment:
Good point. The expected behavior is that JarFileContainer wil be cleaned when the application is cleaned.

This will be tracked by https://github.com/intel-hadoop/gearpump/issues/257